generic_config_updater/field_operation_validators.py

```diff
@@ -1,12 +1,50 @@
-from sonic_py_common import device_info
+import os
 import re
+import json
+import subprocess
+from sonic_py_common import device_info
+from .gu_common import GenericConfigUpdaterError
+
+SCRIPT_DIR = os.path.dirname(os.path.realpath(__file__))
+GCU_TABLE_MOD_CONF_FILE = f"{SCRIPT_DIR}/gcu_field_operation_validators.conf.json"
+
+def get_asic_name():
+    asic = "unknown"
+    
+    if device_info.get_sonic_version_info()['asic_type'] == 'cisco-8000':
+        asic = "cisco-8000"
+    elif device_info.get_sonic_version_info()['asic_type'] == 'mellanox':
+        GET_HWSKU_CMD = "sonic-cfggen -d -v DEVICE_METADATA.localhost.hwsku"
+        spc1_hwskus = [ 'ACS-MSN2700', 'ACS-MSN2740', 'ACS-MSN2100', 'ACS-MSN2410', 'ACS-MSN2010', 'Mellanox-SN2700', 'Mellanox-SN2700-D48C8' ]
+        proc = subprocess.Popen(GET_HWSKU_CMD, shell=True, universal_newlines=True, stdout=subprocess.PIPE)
+        output, err = proc.communicate()
+        hwsku = output.rstrip('\n')
+        if hwsku.lower() in [spc1_hwsku.lower() for spc1_hwsku in spc1_hwskus]:
+            asic = "spc1"
+    elif device_info.get_sonic_version_info()['asic_type'] == 'broadcom':
+        command = ["sudo", "lspci"]
```

Comment:
>command [](http://example.com/codeflow?start=8&length=7)

Runtime check is heavy. We could build the mapping offline, and move add the hareware skus into gcu_field_operation_validators.conf.json.

Then in runtime, you just check hwsku.

Reply:
Moved mapping to gcu_field_operation_validators.conf.json.